modules/screensharing/index.js

```diff
@@ -1,25 +1,48 @@
 const electron = require("electron");
+
 /**
- * Adds screen sharing support. Prompts the user to choose screen, window or
- * tab to share (not implemented yet). And passes the chosen source to caller
- * of the method.
- * @param {Function} callback the success callback
- * @param {Function} errorCallback the callback for errors
+ * Get sources available for screensharing. The callback is invoked
+ * with an array of DesktopCapturerSources.
+ *
+ * @param {Function} callback - The success callback.
+ * @param {Function} errorCallback - The callback for errors.
+ * @param {Object} options - Additional configuration for getting sources.
+ * @param {Array} options.types - Specify the desktop source types to get,
+ * with valid sources being "window" and "screen".
+ * @param {Object} options.thumbnailSize - Specify how big the preview
+ * images for the sources should be. By default electron will return
+ * images with height and width of 150px.
  */
-function obtainDesktopStream (callback, errorCallback) {
-    //FIXME: add more types
-    electron.desktopCapturer.getSources({types: ['screen']},
+function obtainDesktopStreams(callback, errorCallback, options = {}) {
+    electron.desktopCapturer.getSources(options,
         (error, sources) => {
             if (error) {
                 errorCallback(error);
                 return;
             }
-            //FIXME: Implement window picker instead of choosing always the
-            // first sourceId
-            callback(sources[0].id);
+
+            callback(sources);
         });
 }
 
+/**
+ * Adds screen sharing support. Prompts the user to choose screen, window or
+ * tab to share (not implemented yet). And passes the chosen source to caller
+ * of the method.
+ * @param {Function} callback the success callback
+ * @param {Function} errorCallback the callback for errors
+ */
+function obtainDesktopStream(callback, errorCallback) {
+    obtainDesktopStreams(
+        (sources) => callback(sources[0].id),
+        errorCallback,
+        { types: ['screen'] }
+    );
+}
+
 module.exports = function setupScreenSharingForWindow(pWindow) {
-    pWindow.JitsiMeetElectron = {obtainDesktopStream};
+    pWindow.JitsiMeetElectron = {
+        obtainDesktopStream,
```

Comment:
The main reason I exposed both was for backwards compatibility. I'm not completely confident on how far to support backwards compatibility between jitsi-meet, lib-jitsi-meet, and jitsi-meet-electron. I wanted to protect interactions between newer and older versions of lib-jitsi-meet with newer and older versions of the electron app. I was also concerned about the case where a custom jitsi-meet implementation does not have an implemented desktop picker, in which case it felt easier to have another method exposed that auto-selects a DesktopCapturerSource--easier because that is the existing implementation that has been working.